pkg/distributor/validate.go

```diff
@@ -42,12 +42,13 @@ var (
 	reasonTooFarInFuture               = globalerror.SampleTooFarInFuture.LabelValue()
 
 	// Discarded exemplars reasons.
-	reasonExemplarLabelsMissing    = globalerror.ExemplarLabelsMissing.LabelValue()
-	reasonExemplarLabelsTooLong    = globalerror.ExemplarLabelsTooLong.LabelValue()
-	reasonExemplarTimestampInvalid = globalerror.ExemplarTimestampInvalid.LabelValue()
-	reasonExemplarLabelsBlank      = "exemplar_labels_blank"
-	reasonExemplarTooOld           = "exemplar_too_old"
-	reasonExemplarTooFarInFuture   = "exemplar_too_far_in_future"
+	reasonExemplarLabelsMissing           = globalerror.ExemplarLabelsMissing.LabelValue()
+	reasonExemplarLabelsTooLong           = globalerror.ExemplarLabelsTooLong.LabelValue()
+	reasonExemplarTimestampInvalid        = globalerror.ExemplarTimestampInvalid.LabelValue()
+	reasonExemplarLabelsBlank             = "exemplar_labels_blank"
+	reasonExemplarTooOld                  = "exemplar_too_old"
+	reasonExemplarTooFarInFuture          = "exemplar_too_far_in_future"
+	reasonExamplarTooManyExamplerPerSerie = "too_many_exemplars_per_serie"
```

Comment:
Good idea

Reply:
here the reason is actually too many exemplars per serie compare to the limit, do you think the metrics cortex_discarded_exemplars_total with reason max_exemplars_per_series_per_request is straight forward? because the metrics track how many exemplars discarded caused by limit reached.

Reply:
this number = limit - allwed exemplars per serie